onedal/tests/utils/_dataframes_support.py

```diff
@@ -0,0 +1,82 @@
+# ===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import pytest
+
+try:
+    import dpctl
+    import dpctl.tensor as dpt
+
+    dpctl_available = True
+except ImportError:
+    dpctl_available = False
+
+try:
+    import dpnp
+
+    dpnp_available = True
+except ImportError:
+    dpnp_available = False
+
+import numpy as np
+
+from onedal.tests.utils._device_selection import get_queues
+
+
+def _get_dataframes_and_queues(
+    dataframe_filter_="numpy,dpnp,dpctl", device_filter_="cpu,gpu"
+):
+    dataframes_and_queues = [
+        pytest.param("numpy", None, id="numpy"),
+    ]
+
+    def _get_df_and_q(dataframe: str):
+        for queue in get_queues(device_filter_):
+            id = "{}-SyclQueue_{}".format(
+                dataframe, "GPU" if queue.sycl_device.is_gpu else "CPU"
+            )
+            dataframes_and_queues.append(pytest.param(dataframe, queue, id=id))
```

Comment:
Bad practice in this specific case: it's better to return list of new data frames and queues than update of existing `dataframes_and_queues` var. from higher scope

Reply:
this is inner function, that uses or updates function variable. It doesn't makes for them, but I can update it with return values

Reply:
done